CRM/Report/Form/ActivitySummary.php

```diff
@@ -305,6 +313,12 @@ public function from($durationMode = FALSE) {
     $assigneeID = CRM_Utils_Array::key('Activity Assignees', $activityContacts);
     $targetID = CRM_Utils_Array::key('Activity Targets', $activityContacts);
     $sourceID = CRM_Utils_Array::key('Activity Source', $activityContacts);
+    $activityContactTables = array(
+      $assigneeID => 'assignment_activity',
+      $targetID => 'target_activity',
+      $sourceID => 'source_activity',
+    );
+    $activityRole = $activityContactTables[$this->_params['activity_role_value']];
```

Comment:
Maybe replace this with the following:

```php
$activityContactJoins = '';
foreach ($activityContactTables as $rId => $rAlias) {
  $activityContactJoins .= <<<HERESQL
                LEFT JOIN civicrm_activity_contact $ralias
                       ON {$this->_aliases['civicrm_activity']}.id = $ralias.activity_id AND
                          $rAlias.record_type_id = $rId
HERESQL;
}
$activityContactJoins .= <<<HERESQL
                LEFT JOIN civicrm_contact contact_civireport
                       ON {$activityContactTables[$this->_params['activity_role_value']]}.contact_id = contact_civireport.id
HERESQL;
```

and then replace the first four left joins with `{$activityContactJoins}`

Reply:
Forget it--it ends up being equally long!